### PR TITLE
Refactor to remove item from cart client side

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Remove server-side logic related to removing line item from cart.

Remove useForm hook.

Move transform from API to CartPage props client-side.

Add hook to get current shop session based on cookies.

## Justify why they are needed

Simplify the setup, being able to co-locate more.

I realizing I'm back-tracking a bit but I actually started to like this setup!
If we are using Apollo Client, let's lean into it.

We also get the benefit of using the apollo dev tools.

AND most importantly, this speeds up the UX since there's no serverless function involved.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #444** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/444" target="_blank">Graphite</a> 
    * **PR #445** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/445" target="_blank">Graphite</a> 
      * **PR #447** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/447" target="_blank">Graphite</a>  👈
        * **PR #448** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/448" target="_blank">Graphite</a> 
          * **PR #449** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/449" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/447?utm_source=stack-comment).

File: apps/store/src/services/shopSession/ShopSession.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Cookies from 'js-cookie'
+import { useCallback } from 'react'
+import { useShopSessionLazyQuery } from '@/services/apollo/generated'
+import { COOKIE_KEY_SHOP_SESSION } from './ShopSession.constants'
+
+export const useGetCurrentShopSession = () => {
+  const [getShopSession] = useShopSessionLazyQuery()
+
+  return useCallback(async () => {
+    const shopSessionId = Cookies.get(COOKIE_KEY_SHOP_SESSION)
```
Comment (alebedev) [inline apps/store/src/services/shopSession/ShopSession.hooks.ts:10]: Are we now reading cookies directly instead of using ShopSessionService?

**Review by alebedev — APPROVED** [review]
Looks simpler than it was, this is good thing

File: apps/store/src/services/shopSession/ShopSession.hooks.ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/services/shopSession/ShopSession.hooks.ts:10]: Hmm yes agreed, that's a lazy shortcut 😅

We should pass this ID from the server in this case. I will update in an upcoming PR!

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/447?utm_source=stack-comment).


Job Reference: bYOGtcBHRMqrhfixkOY3

**Pull request merged**